client/src/components/AppCommandBar/AppCommandBar.test.js

```diff
@@ -0,0 +1,171 @@
+    showProp: "showOutput"
+  }
+];
+
+let toggleCounter = 0;
+
+const toggleOptionalComponent = () => {
+  toggleCounter++;
+};
+
+let container = null;
+beforeEach(() => {
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  PubSub.clearAllSubscriptions();
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+
+test("delete all twins component", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    apiService.getAllTwins.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".delete-button");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    const deleteButton = await screen.findByTestId("deleteTwins");
+
+    act(() => {
+      deleteButton.dispatchEvent(new MouseEvent("click", { bubbles: true }))
+    });
+
+    expect(getAllTwins).toHaveBeenCalledTimes(1);
+
+  });
+
+  test("switch environment", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    configService.setConfig.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
+        optionalComponentsState={optionalComponentsState}/>, container);});
+    
+    const button = container.querySelector(".sign-in");
+    act(() => {
+      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+    });
+
+    let input = await screen.findByTestId("urlInput");
+    input.value = "https://test.api.scus.digitaltwins.azure.net";
+    Simulate.change(input);
+    expect(input.value).toBe("https://test.api.scus.digitaltwins.azure.net");
+    const saveButton = await screen.findByTestId("saveConfiguration");
+    act(() => {
+      saveButton.dispatchEvent(new MouseEvent("click", { bubbles: true }));
+    });
+
+    expect(setConfig).toHaveBeenCalledTimes(1);
+    await waitFor(() => expect(eventService.publishEnvironmentChange).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(eventService.publishConfigure).toHaveBeenCalledTimes(1));
+    
+  });
+
+  test("switch console", async () => {
+    // The <ModelViewerComponent /> component calls the config service and won't call the API unless the appAdtUrl is set
+    configService.getConfig.mockResolvedValue({ appAdtUrl: "https://foo" });
+    configService.setConfig.mockResolvedValue(mockSuccesResponse);
+
+    act(() => {render(<AppCommandBar optionalComponents={optionalComponentsState}
```

Comment:
I'm not sure if optionalComponents is actually used by the AppCommandBar. Might be an opportunity to remove it?